src/gsi/cplr_wrwrfmassa.f90

```diff
@@ -2200,18 +2261,158 @@ subroutine wrwrfmassa_netcdf_wrf(this,mype)
+
+                   if ( l_use_cvpqx ) then
+                      if ( l_use_cvpqx_pval .eq. 0. ) then ! CVlogq
+                         if (ges_tsen(j,i,k,it) .gt. 274.15) then
+                            qr_min=2.9E-6_r_kind
+                            qr_thrshd=qr_min * one_tenth
+                            qs_min=0.1E-9_r_kind
+                            qs_thrshd=qs_min
+                            qg_min=3.1E-7_r_kind
+                            qg_thrshd=qg_min * one_tenth
+                         else if (ges_tsen(j,i,k,it) .le. 274.15 .and. ges_tsen(j,i,k,it) .ge. 272.15) then
+                            qr_min=2.0E-6_r_kind
+                            qr_thrshd=qr_min * one_tenth
+                            qs_min=1.3E-7_r_kind
+                            qs_thrshd=qs_min * one_tenth
+                            qg_min=3.1E-7_r_kind
+                            qg_thrshd=qg_min * one_tenth
+                         else if (ges_tsen(j,i,k,it) .lt. 272.15) then
+                            qr_min=0.1E-9_r_kind
+                            qr_thrshd=qr_min
+                            qs_min=6.3E-6_r_kind
+                            qs_thrshd=qs_min * one_tenth
+                            qg_min=3.1E-7_r_kind
+                            qg_thrshd=qg_min * one_tenth
+                         end if
+!                        convert log(qr/qs/qg) back to qr/qs/qg
+                         if (mype==0 .AND. i==3 .AND. j==3 .AND. k==3) then
+                            write(6,*)'wrwrfmass_netcdf: convert log(qr/qs/qg) back to qr/qs/qg.'
+                            write(6,*)'wrwrfmass_netcdf: then reset (qr/qs/qg) to 0.0 for some cases.'
+                         end if
+
+                         qr_tmp=exp(ges_qr(j,i,k))
+                         qs_tmp=exp(ges_qs(j,i,k))
+                         qg_tmp=exp(ges_qg(j,i,k))
+
+!                        if no update or very tiny value of qr/qs/qg, re-set/clear it off to zero
+                         if ( abs(qr_tmp - qr_min) .lt. (qr_min*r0_01) ) then
+                            qr_tmp=zero
+                         else if (qr_tmp .lt. qr_thrshd) then
+                            qr_tmp=zero
+                         end if
+
+                         if ( abs(qs_tmp - qs_min) .lt. (qs_min*r0_01) ) then
+                            qs_tmp=zero
+                         else if (qs_tmp .lt. qs_thrshd) then
+                            qs_tmp=zero
+                         end if
+
+                         if ( abs(qg_tmp - qg_min) .lt. (qg_min*r0_01) ) then
+                            qg_tmp=zero
+                         else if (qg_tmp .lt. qg_thrshd) then
+                            qg_tmp=zero
+                         end if
+
+                      else if ( l_use_cvpqx_pval .gt. 0. ) then   ! CVpq
+!                        convert log(qr/qs/qg) back to qr/qs/qg
+                         if (mype==0 .AND. i==3 .AND. j==3 .AND. k==3) then
```

Comment:
convert this kind of write to comments or move them out of the loop.

Reply:
I modified the code according to your comments. I took out these codes from loops and make them print one time at the start of the routine using if(mype==0).

https://github.com/NOAA-EMC/GSI/pull/142/files#diff-83013677e0fb286b5f115ca5766f4b67266ed22b3d70b8867ed865db67ab8f50R3213-R3226